### PR TITLE
Use base URL provided by Python backend (or mocked if offline) instead of building via JS

### LINKED ISSUE #343: Revisit how we build the GraphQL service URL
**Describe exactly what you would like to see in an upcoming release**

Some JS applications have a fixed URL for the backend (e.g. github.com). Other APIs require users to change a file during deployment (e.g. alter `dist/config.js` and set the URL). In our case, we are making a few assumptions for now.

We use the `window.location` [API](https://developer.mozilla.org/en-US/docs/Web/API/Location), but even though there is a spec, it is a big vague on slashes, and in the end it is still up to browsers to implement it correctly.

This has worked well, but mainly for being development only. Looking at how users will deploy and use the UI Server and the UI applications may give better idea on how to implement this.

**Additional context**

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by kinow**
These changes close #343 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Comment by kinow** [comment]
Requires a change on the server side / backend to have the base URL populated.

**Comment by kinow** [comment]
The e2e tests are failing due to a bug fixed in the Yarn PR. Not sure why it only appeared now :bug: 

The test failure is also affecting @oliver-sanders 's PR for isHeld.

File: public/index.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <noscript>
       <strong>We're sorry but Cylc UI doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
     </noscript>
-    <div id="app"></div>
+    <div id="app" data-base-url="{{ python_base_url }}"></div>
```
Comment (oliver-sanders) [inline public/index.html:14]: Where does `python_base_url` come from?

Comment (kinow) [reply public/index.html:14]: From Cylc UI Server. The `index.html` is rendered by Tornado. In the offline mode this value will be as-is, but it is ignored in JS.

**Comment by kinow** [comment]
Rebased.

File: public/index.html (hunk 1 above)
Comment (oliver-sanders) [reply public/index.html:14]: Aaah

Comment (kinow) [reply public/index.html:15]: Added a comment explaining the above :+1:

**Comment by oliver-sanders** [comment]
Had a go at running:

```console
$ npm run serve
```

Untitled

**Pull request merged**